sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -233,12 +233,11 @@ private[sql] case class PreWriteCheck(conf: SQLConf, catalog: SessionCatalog)
         }
 
         PartitioningUtils.validatePartitionColumn(
-          c.child.schema, c.partitionColumns, conf.caseSensitiveAnalysis)
+          query.schema, tableDesc.partitionColumnNames, conf.caseSensitiveAnalysis)
 
         for {
-          spec <- c.bucketSpec
-          sortColumnName <- spec.sortColumnNames
-          sortColumn <- c.child.schema.find(_.name == sortColumnName)
+          spec <- tableDesc.bucketSpec
+          sortColumn <- tableDesc.schema.filter(spec.sortColumnNames.contains)
```

Comment:
Below is the logics for bucketing tables. If we do not plan to support Hive bucketing tables, maybe we just issue an exception?